src/main/java/com/google/api/core/ApiFutures.java

```diff
@@ -77,7 +77,8 @@ public void onSuccess(V v) {
         Futures.catching(
             listenableFutureForApiFuture(input),
             exceptionType,
-            new GaxFunctionToGuavaFunction<X, V>(callback));
+            new GaxFunctionToGuavaFunction<X, V>(callback),
+            directExecutor());
```

Comment:
Add a version of catching that takes an executor, mark this method `@Deprecated`